### DISCUSSION TITLE
RuntimeError: This event loop is already running

### DISCUSSION THREAD

**Discussion by luciameng1989** [Future-House/paper-qa#125] [Q&A] [answered]
I'm trying to use locally hosted GPT4ALL-j model and locally hosted huggingfaceembeddings model to run the code. Everything works fine till the line to query from document, I got an error:

RuntimeError: This event loop is already running

Any ideas?

**Comment by whitead** [comment, marked as answer]
Are you in a jupyter notebook? If so, add this to the top of your notebook

```py
import nest_asyncio
nest_asyncio.apply()
```
